codebuild/spec/buildspec_analyzers.yml

```diff
@@ -39,76 +39,59 @@ version: 0.2
 # in the stack traces that ASAN reports.
 batch:
   build-list:
-    - identifier: clang_awslc
+    - identifier: clang_awslc_sanitizers
+      buildspec: codebuild/spec/buildspec_sanitizers.yml
       env:
         compute-type: BUILD_GENERAL1_LARGE
         variables:
           S2N_LIBCRYPTO: awslc
           COMPILER: clang
-    - identifier: clang_openssl_3_0
+    - identifier: clang_openssl_3_0_sanitizers
+      buildspec: codebuild/spec/buildspec_sanitizers.yml
       env:
         compute-type: BUILD_GENERAL1_LARGE
         variables:
           S2N_LIBCRYPTO: openssl-3.0
           COMPILER: clang
-    - identifier: clang_openssl_1_1_1
+    - identifier: clang_openssl_1_1_1_sanitizers
+      buildspec: codebuild/spec/buildspec_sanitizers.yml
       env:
         compute-type: BUILD_GENERAL1_LARGE
         variables:
           S2N_LIBCRYPTO: openssl-1.1.1
           COMPILER: clang
-    - identifier: clang_openssl_1_0_2
+    - identifier: clang_openssl_1_0_2_sanitizers
+      buildspec: codebuild/spec/buildspec_sanitizers.yml
       env:
         compute-type: BUILD_GENERAL1_LARGE
         variables:
           S2N_LIBCRYPTO: openssl-1.0.2
           COMPILER: clang
-    - identifier: gcc_awslc
+    - identifier: gcc_awslc_sanitizers
+      buildspec: codebuild/spec/buildspec_sanitizers.yml
       env:
         compute-type: BUILD_GENERAL1_LARGE
         variables:
           S2N_LIBCRYPTO: awslc
           COMPILER: gcc
-    - identifier: gcc_openssl_3_0
+    - identifier: gcc_openssl_3_0_sanitizers
+      buildspec: codebuild/spec/buildspec_sanitizers.yml
       env:
         compute-type: BUILD_GENERAL1_LARGE
         variables:
           S2N_LIBCRYPTO: openssl-3.0
           COMPILER: gcc
-    - identifier: gcc_openssl_1_1_1
+    - identifier: gcc_openssl_1_1_1_sanitizers
```

Comment:
Hmm, I'm actually gonna vote to remove `gcc_openssl_1_1_1_sanitizers` and `gcc_openssl_1_0_2_sanitizers`.

My reasoning is that
- clang is the main/more advanced compiler for ASAN and UBSAN coverage, gcc coverage is just a potential extra
- openssl 1.0.2 and 1.1.1 are already expired, so the extra coverage on these is overkill. We want _some_ amount of coverage, but they are no longer happy path options and don't deserve the deeper inspection.

The reason I'm picking on this now is because we're about to add a _lot_ more jobs to this buildspec, so we want to avoid an unnecessary bloat.